docs(docs-infra): update service worker browser compatibility reference (#51736)
angular/angular: aio/content/guide/service-worker-intro.md

For more information about service workers in general, see [Service Workers: an Introduction](https://developers.google.com/web/fundamentals/primers/service-workers).

> For more information about browser support, see [MDN](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker)'s [Browser Compatibility](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker#browser_compatibility) section of [ServiceWorker](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker), Jake Archibald's [Is Serviceworker ready?](https://jakearchibald.github.io/isserviceworkerready), and [Can I Use](https://caniuse.com/serviceworkers).

Reviewer: ```suggestion
For more information about browser support, see MDN's [Service Worker Browser Compatibility](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker#browser_compatibility) and [Can I Use](https://caniuse.com/serviceworkers).
```

We can drop Jake's link here as SW is shipped across all the browsers he was tracking anyway.
